google/cloud/internal/async_streaming_write_rpc.h

```diff
@@ -0,0 +1,121 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef GOOGLE_CLOUD_CPP_GOOGLE_CLOUD_INTERNAL_ASYNC_STREAMING_WRITE_RPC_H
+#define GOOGLE_CLOUD_CPP_GOOGLE_CLOUD_INTERNAL_ASYNC_STREAMING_WRITE_RPC_H
+
+#include "google/cloud/future.h"
+#include "google/cloud/internal/grpc_request_metadata.h"
+#include "google/cloud/status_or.h"
+#include "google/cloud/version.h"
+
+namespace google {
+namespace cloud {
+GOOGLE_CLOUD_CPP_INLINE_NAMESPACE_BEGIN
+namespace internal {
+
+/**
+ * An abstraction for asynchronous streaming write RPCs.
+ *
+ * Streaming write RPCs (sometimes called client-side streaming RPCs) allow
+ * applications to send multiple "requests" on the same RPC.  They are often
+ * used in services where:
+ *
+ * - The data sent to the service is large, such as uploads or bulk inserts.
+ * - Sending the data in small RPCs would be too slow, as each would require
+ *   a full round trip to the service.
+ *
+ * @note Objects of this type should not be destroyed until the future returned
+ *     by `Finish()` is satisfied.
+ */
+template <typename Request, typename Response>
+class AsyncStreamingWriteRpc {
+ public:
+  virtual ~AsyncStreamingWriteRpc() = default;
+
+  /**
+   * Sends a best-effort request to cancel the RPC.
+   *
+   * The library code should still wait for the current operation(s) (any
```

Comment:
s/operations(s) (any/operation (a/

as there can only be one pending operation at a time. (I think...)

Reply:
Done